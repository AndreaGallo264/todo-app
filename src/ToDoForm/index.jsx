import React, { useState } from "react";

const ToDoForm = ({ addTask }) => {
  const [userInput, setUserInput] = useState({
    task: "",
    dueDate: new Date(),
  });
  const handleChange = (event) => {
    setUserInput({
      ...userInput,
      [event.target.name]: event.target.value,
    });
  };
  const handleSubmit = (e) => {
    e.preventDefault();
    addTask(userInput);
    setUserInput({
      task: "",
      dueDate: new Date(),
    });
  };
  return (
    <form onSubmit={handleSubmit}>
      <input
        value={userInput.task}
        type="text"
        onChange={handleChange}
        placeholder="Enter task..."
        name="task"
      />
      <input
        type="date"
        id="start"
        name="dueDate"
        value="2018-07-22"
        min="2018-01-01"
        max="2018-12-31"
        onChange={handleChange}
      />
      <button type="submit">Add</button>
    </form>
  );
};

export default ToDoForm;
