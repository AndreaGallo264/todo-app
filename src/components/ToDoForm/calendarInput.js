import React, { useState, useEffect } from "react";
import MomentUtils from "@date-io/moment";
import {
  MuiPickersUtilsProvider,
  KeyboardDatePicker,
} from "@material-ui/pickers";
import { formatDateExpression } from "../Utils/formatDate";

const CalendarInput = ({ selectedDate, handleDateChange }) => {
  const [valueDate, setDate] = useState(selectedDate);

  useEffect(() => {
    setDate(selectedDate);
  }, [selectedDate]);

  return (
    <MuiPickersUtilsProvider utils={MomentUtils}>
      <KeyboardDatePicker
        disableToolbar
        variant="inline"
        format={formatDateExpression}
        margin="normal"
        label="Due date"
        value={valueDate}
        onChange={handleDateChange}
        KeyboardButtonProps={{
          "aria-label": "change date",
        }}
        required
        name="dueDate"
        asp-format={`0:${formatDateExpression}`}
      />
    </MuiPickersUtilsProvider>
  );
};

export default CalendarInput;
