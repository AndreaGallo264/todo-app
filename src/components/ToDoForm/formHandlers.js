import Swal from "sweetalert2";
import { formatDate } from "../Utils/formatDate";
import { addTask, incrementId } from "../../store/actions";

export const handleChange = (event, userInput, setUserInput) => {
  setUserInput({ ...userInput, [event.target.name]: event.target.value });
};

export const handleDateChange = (event, userInput, setUserInput) => {
  if (event) {
    if (!event._d.toString().includes("Invalid")) {
      const formatedDate = formatDate(event._d);
      setUserInput({ ...userInput, dueDate: formatedDate });
    }
  } else {
    setUserInput({ ...userInput, dueDate: "" });
  }
};

export const handleSubmit = (
  event,
  userInput,
  dispatch,
  setUserInput,
  initialValue,
  history
) => {
  event.preventDefault();
  if (!userInput.dueDate.includes("Invalid") && userInput.dueDate.length > 0) {
    dispatch(incrementId());
    dispatch(addTask(userInput));
    setUserInput(initialValue);
    successAlert();
    history.push("/");
  }
};

const successAlert = () => {
  Swal.fire({
    icon: "success",
    title: "Success!",
    text: "Your task was successfully created!",
    showConfirmButton: false,
    timer: 2000,
  });
};
