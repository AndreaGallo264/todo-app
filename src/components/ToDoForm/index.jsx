import React, { useState } from "react";
import { Button, TextField, Grid, Typography } from "@material-ui/core";

//Components
import CalendarInput from "./calendarInput";
import { formatDate } from "../Utils/formatDate";

const ToDoForm = ({ addTask }) => {
  const today = formatDate("");
  const initialValue = { task: "", dueDate: today };
  const [userInput, setUserInput] = useState(initialValue);

  const handleChange = (event) => {
    setUserInput({ ...userInput, [event.target.name]: event.target.value });
  };

  const handleDateChange = (event) => {
    const formatedDate = formatDate(event._d);
    setUserInput({ ...userInput, dueDate: formatedDate });
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    //Checks if date is invalid
    if (!userInput.dueDate.includes("Invalid")) {
      addTask(userInput);
      setUserInput(initialValue);
    }
  };
  return (
    <>
      <Typography variant="h3" color="primary">
        Create a new task
      </Typography>
      <form onSubmit={handleSubmit}>
        <Grid container item xs={12} md={6} justify="center" direction="column">
          <TextField
            value={userInput.task}
            onChange={handleChange}
            label="Task"
            name="task"
            required
          />
          <CalendarInput
            selectedDate={userInput.dueDate}
            handleDateChange={handleDateChange}
          />
          <Button type="submit" variant="contained" color="primary">
            Add
          </Button>
        </Grid>
      </form>
    </>
  );
};

export default ToDoForm;
