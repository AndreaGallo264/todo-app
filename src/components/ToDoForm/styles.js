import { makeStyles } from "@material-ui/core";
export const useStyles = makeStyles({
  media: {
    height: "15rem",
    width: "40rem",
    maxWidth: "100%",
  },
  inputContainer: {
    display: "flex",
    flexDirection: "column",
  },
  addBtn: {
    width: "100%",
  },
  formClases: { width: "35rem", maxWidth: "100%" },
});
