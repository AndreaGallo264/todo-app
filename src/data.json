[{
  "id": 1,
  "task": "Give dog a bath",
  "complete": true,
  "dueDate": "2021-05-08"
}, {
  "id": 2,
  "task": "Do laundry",
  "complete": true,
  "dueDate": "2021-05-08"
}, {
  "id": 3,
  "task": "Vacuum floor",
  "complete": false,
  "dueDate": "2021-05-08"
}, {
  "id": 4,
  "task": "Feed cat",
  "complete": true,
  "dueDate": "2021-05-08"
}, {
  "id": 5,
  "task": "Change light bulbs",
  "complete": false,
  "dueDate": "2021-05-08"
}, {
  "id": 6,
  "task": "Go to Store",
  "complete": true,
  "dueDate": "2021-05-08"
}, {
  "id": 7,
  "task": "Fill gas tank",
  "complete": true,
  "dueDate": "2021-05-08"
}, {
  "id": 8,
  "task": "Change linens",
  "complete": false,
  "dueDate": "2021-05-08"
}, {
  "id": 9,
  "task": "Rake leaves",
  "complete": true,
  "dueDate": "2021-05-08"
}, {
  "id": 10,
  "task": "Bake Cookies",
  "complete": false,
  "dueDate": "2021-05-08"
}, {
  "id": 11,
  "task": "Take nap",
  "complete": true,
  "dueDate": "2021-05-08"
}, {
  "id": 12,
  "task": "Read book",
  "complete": true,
  "dueDate": "2021-05-08"
}, {
  "id": 13,
  "task": "Exercise",
  "complete": false,
  "dueDate": "2021-05-08"
}, {
  "id": 14,
  "task": "Give dog a bath",
  "complete": false,
  "dueDate": "2021-05-08"
}, {
  "id": 15,
  "task": "Do laundry",
  "complete": false,
  "dueDate": "2021-05-08"
}, {
  "id": 16,
  "task": "Vacuum floor",
  "complete": false,
  "dueDate": "2021-05-08"
}, {
  "id": 17,
  "task": "Feed cat",
  "complete": true,
  "dueDate": "2021-05-08"
}, {
  "id": 18,
  "task": "Change light bulbs",
  "complete": false,
  "dueDate": "2021-05-08"
}, {
  "id": 19,
  "task": "Go to Store",
  "complete": false,
  "dueDate": "2021-05-08"
}, {
  "id": 20,
  "task": "Fill gas tank",
  "complete": false,
  "dueDate": "2021-05-08"
}]