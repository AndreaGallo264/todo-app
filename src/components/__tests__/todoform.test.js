import React from "react";
import renderer from "react-test-renderer";
import configureMockStore from "redux-mock-store";
import { Provider } from "react-redux";
import data from "../../data.json";
import ToDoForm from "../ToDoForm";

describe("ToDoForm component", () => {
  test("Matches snapshot", () => {
    const mockStore = configureMockStore();
    const store = mockStore({ list: data, lastId: data.length });
    const tree = renderer
      .create(
        <Provider store={store}>
          <ToDoForm />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
