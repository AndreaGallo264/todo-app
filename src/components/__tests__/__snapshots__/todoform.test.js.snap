// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToDoForm component Matches snapshot 1`] = `
<div
  className="MuiBox-root MuiBox-root-5"
>
  <h3
    className="MuiTypography-root MuiTypography-h3 MuiTypography-colorPrimary"
  >
    Create new
  </h3>
  <form
    className="makeStyles-formClases-4"
    onSubmit={[Function]}
  >
    <div
      className="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        className="MuiCardMedia-root makeStyles-media-1"
        style={
          Object {
            "backgroundImage": "url(\\"formImg.jpg\\")",
          }
        }
        title="Contemplative Reptile"
      />
      <div
        className="MuiCardContent-root makeStyles-inputContainer-2"
      >
        <div
          className="MuiFormControl-root MuiTextField-root"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
            data-shrink={false}
          >
            Task
            <span
              aria-hidden={true}
              className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiInput-input"
              disabled={false}
              name="task"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          asp-format="0:yyyy-MM-DD"
          className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled Mui-required Mui-required"
            data-shrink={true}
          >
            Due date
            <span
              aria-hidden={true}
              className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd"
              disabled={false}
              name="dueDate"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              type="text"
              value="2021-05-08"
            />
            <div
              className="MuiInputAdornment-root MuiInputAdornment-positionEnd"
            >
              <button
                aria-label="change date"
                className="MuiButtonBase-root MuiIconButton-root"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                <span
                  className="MuiIconButton-label"
                >
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                    />
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                  </svg>
                </span>
              </button>
            </div>
          </div>
          
        </div>
      </div>
      <div
        className="MuiCardActions-root MuiCardActions-spacing"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-addBtn-3 MuiButton-containedPrimary"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="submit"
        >
          <span
            className="MuiButton-label"
          >
            Add
          </span>
        </button>
      </div>
    </div>
  </form>
</div>
`;
